### DISCUSSION TITLE
Why stm32 rtc doesn't allow time before year 2000

### DISCUSSION THREAD

**Discussion by Martdur** [zephyrproject-rtos/zephyr#74023] [Q&A] [answered]
Trying to implement a timestamp from 0 to the end of life of the product I was using the RTC in order to keep counting when powered only with Vbatt.
But I can't set the time to 0, and can not read without calling `rtc_set_time`.
Is there a way to write to the rtc register the time value in order to not set the time at run time ?
Why can I set the time to 0 ?

**Comment by ajarmouni-st** [comment, marked as answer]
Hi @Martdur it's a hardware limitation, see https://community.st.com/t5/stm32-mcus-products/stm32-rtc-fixed-to-21st-century/m-p/177837/highlight/true#M37132

**Comment by Martdur** [comment]
Hi @ajarmouni-st  Thanks for the answer !

It's indeed a hardware limitation.
I'm using a workaround by setting the minimal possible time and then substract the date time with this value to get a counter from 0.
